src/test/coins_tests.cpp

```diff
@@ -612,7 +612,7 @@ BOOST_AUTO_TEST_CASE(ccoins_access)
     CheckAccessCoins(ABSENT, VALUE2, VALUE2, FRESH      , FRESH      );
     CheckAccessCoins(ABSENT, VALUE2, VALUE2, DIRTY      , DIRTY      );
     CheckAccessCoins(ABSENT, VALUE2, VALUE2, DIRTY|FRESH, DIRTY|FRESH);
-    CheckAccessCoins(PRUNED, ABSENT, PRUNED, NO_ENTRY   , FRESH      );
+    CheckAccessCoins(PRUNED, ABSENT, ABSENT, NO_ENTRY   , NO_ENTRY   );
```

Comment:
Explanation of this change: Calling CCoinsViewCache::AccessCoins when no cache entry exists and the base entry is pruned no longer creates a cache entry.

The changed behavior in this case is better than the current behavior because now this case is consistent with the case in line 606 above where no cache entry exists and no base entry exists. It doesn't make any sense for the cache to behave differently just because a base entry is pruned instead of nonexistent.